Framework/Algorithms/test/RebinTest.h

```diff
@@ -748,6 +812,537 @@ class RebinTest : public CxxTest::TestSuite {
+    // test successful execution
+    TS_ASSERT_THROWS_NOTHING(rebin.execute());
+    TS_ASSERT(rebin.isExecuted());
+    // ensure nothing changed in execution
+    TS_ASSERT(bool(rebin.getProperty("UseReverseLogarithmic")));
+
+    AnalysisDataService::Instance().remove("test_in1D");
+    AnalysisDataService::Instance().remove("test_out1D");
+  }
+
+  void do_test_changes_rebin_params(std::string binMode, double step, double good_step, bool only_step = false) {
+    Workspace2D_sptr test_in1D = Create1DWorkspace(50);
+    AnalysisDataService::Instance().add("test_in1D", test_in1D);
+
+    double start, stop;
+    std::stringstream params;
+    if (only_step) {
+      MatrixWorkspace_sptr inputdata = AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>("test_in1D");
+      auto &outX = inputdata->x(0);
+      start = outX[0];
+      stop = outX[outX.size() - 1];
+      params << step;
+    } else {
+      start = 1.0;
+      stop = 10.0;
+      params << start << "," << step << "," << stop;
+    }
+
+    // check that algo will change rebin params
+    Rebin rebin;
+    rebin.initialize();
+    rebin.setPropertyValue("InputWorkspace", "test_in1D");
+    rebin.setPropertyValue("OutputWorkspace", "test_out1D");
+    rebin.setPropertyValue("Params", params.str());
+    rebin.setPropertyValue("BinningMode", binMode);
+    if (binMode == "Power")
+      rebin.setProperty("Power", 0.5);
+    auto errmsgs = rebin.validateInputs();
+    TS_ASSERT(errmsgs.empty());
+
+    std::vector<double> rbParams = rebin.getProperty("Params");
+    TS_ASSERT(rbParams[0] == start);
+    TS_ASSERT(rbParams[1] == good_step);
+    TS_ASSERT(rbParams[2] == stop);
+
+    // ensure execution, and nothing changes in execution
+    TS_ASSERT_THROWS_NOTHING(rebin.execute());
+    TS_ASSERT(rebin.isExecuted());
+    std::vector<double> rebinAgainParams = rebin.getProperty("Params");
+    TS_ASSERT_EQUALS(rbParams, rebinAgainParams);
+
+    AnalysisDataService::Instance().remove("test_in1D");
+    AnalysisDataService::Instance().remove("test_out1D");
+  }
+
+  void test_all_binning_modes_change_rebin_params() {
+    std::vector<std::string> binModes{"Linear", "Logarithmic", "ReverseLogarithmic", "Power"};
+    std::vector<double> steps{-1.0, 1.0, 1.0, -1.0};
+    std::vector<double> goodSteps{1.0, -1.0, -1.0, 1.0};
+
+    for (size_t i = 0; i < binModes.size(); i++) {
+      for (char only_step = 0; only_step < 2; only_step++) {
+        do_test_changes_rebin_params(binModes[i], steps[i], goodSteps[i], only_step);
+      }
+    }
+  }
+
+  void do_test_changes_power(std::string binMode, double step) {
+    // check that algo run in binning  mode will change power
+    std::stringstream params;
+    params << "1.0," << step << ",100.0";
+
+    Workspace2D_sptr test_in1D = Create1DWorkspace(50);
+    AnalysisDataService::Instance().add("test_in1D", test_in1D);
+
+    Rebin rebin;
+    rebin.initialize();
+    rebin.setPropertyValue("InputWorkspace", "test_in1D");
```

Comment:
Debugging is a little easier if you set the workspace names to be similar to the test name. Also, it can be good to define the names as string constants at the top of the test, e.g.
```c
const std::string IN_WKSP("changes_power_in");
const std::string OUT_WKSP("changes_power_out");
```